algorithms/kernel/k_nearest_neighbors/kdtree_knn_impl.i

```diff
@@ -160,11 +139,27 @@ public:
     }
 
 private:
-    T * _data;
-    size_t _top;
-    size_t _count;
-    size_t _size;
-    size_t _sizeMinus1;
+    T * _data          = nullptr;
+    size_t _top        = 0;
+    size_t _sizeMinus1 = 0;
+    size_t _size       = 0;
+    DAAL_FORCEINLINE void setSize(size_t size)
+    {
+        _size       = size;
+        _sizeMinus1 = static_cast<size_t>(_size - 1);
+    }
+    /*
+        There was a strong assumption that stack can be initialized by only powers of 2.
+        Due to the usage scenario it's true but wrong in general. This function should fix it.
```

Comment:
I think this comment confuses people who are not familiar with previous kNN code.